File: salimmo-parent/property-service/src/main/java/org/propertyservice/dto/PropertyDto.java
package org.propertyservice.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;
import org.propertyservice.enums.PropertyCategory;
import org.propertyservice.enums.PropertyType;

import java.util.UUID;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class PropertyDto {
    private Long id;
    private UUID ref;
    @JsonIgnoreProperties("property")
    private PropertyOwnerDto owner;
    private PropertyCategory category;
    private PropertyType type;
    @JsonIgnoreProperties("property")
    private PropertyLocationDto propertyLocation;
    @JsonIgnoreProperties("property")
    private InnerPropertyDto innerProperty;
    @JsonIgnoreProperties("property")
    private PropertySurfaceDto propertySurface;
    @JsonIgnoreProperties("property")
    private PropertyEnergiesDto propertyEnergies;
    @JsonIgnoreProperties("property")
    private ExteriorPropertyDto exteriorProperty;
    private String title;
    private String description;
    private Double fees;
    private Double netPrice;
    private Byte [] images;
}

File: salimmo-frontend/src/app/models/property/propertyDto.ts
import {PropertyOwnerDto} from "../propertyowner/propertyOwnerDto";
import {PropertyCategory} from "../enums/property-category";
import {PropertyType} from "../enums/property-type";
import {PropertyLocationDto} from "../propertylocation/propertyLocationDto";
import {InnerPropertyDto} from "../innerproperty/innerPropertyDto";
import {PropertySurfaceDto} from "../propertysurface/propertySurfaceDto";
import {PropertyEnergiesDto} from "../propertyenergies/propertyEnergiesDto";
import {ExteriorPropertyDto} from "../exteriorproperty/exteriorPropertyDto";
import {PropertyStatus} from "../enums/property-status";

export class PropertyDto {
  id!:number;
  ref!:string;
  owner!:PropertyOwnerDto;
  category!:PropertyCategory;
  type!:PropertyType;
  propertyLocation!:PropertyLocationDto;
  innerProperty!:InnerPropertyDto;
  propertySurface!:PropertySurfaceDto;
  propertyEnergies!:PropertyEnergiesDto;
  exteriorProperty!:ExteriorPropertyDto;
  title!:string;
  description!:string;
  fees!:number;
  netPrice!:number;
  images!:string[];
  status!:PropertyStatus;
  createdAt!:Date;
}
